mime_types.go

```diff
@@ -0,0 +1,20 @@
+package main
+
+import (
+	"log"
+	"mime"
+)
+
+func init() {
+	mustAddMimeExtenstionType(".gz", "application/gzip")
```

Comment:
LGTM. Would be nice to have a note in here explaining why we define it here with all the things you were digging up.